### PR TITLE
docs: Add documentation for the new Transcribe method.

### PR DESCRIPTION
**Pull request by mplooster**
Added descriptions for the new Transcribe method and supporting
Transcribe action. Included examples for Go, Python, and C++.
Also updated the Python StreamASR convenience function example to
use the updated client based method.

Regenerated the protobuf API docs to reflect the latest protobuf
file, and fixed a typo in the protobuf comments that was manifest
in the autogenerated file.

### CODE REVIEW

File: docs-src/content/audio-input/_index.md
Comment (renaissanceGeek) [inline docs-src/content/audio-input/_index.md:230]:
```suggestion
such as taking a note, sending a text message, etc. This stream
```

File: docs-src/content/audio-input/_index.md
Comment (renaissanceGeek) [inline docs-src/content/audio-input/_index.md:334]:
```suggestion
Note that multiple non-partial results may be received by the
```

File: docs-src/content/audio-input/_index.md
Comment (renaissanceGeek) [inline docs-src/content/audio-input/_index.md:337]:
```suggestion
Due to the bidirectional nature of the Transcribe stream, results must
```

File: docs-src/content/audio-input/_index.md
Comment (renaissanceGeek) [inline docs-src/content/audio-input/_index.md:370]: I think it might be a more realistic and helpful example if we check for IsPartial and do something a bit different. This applies to all the language examples, but I'll use the go example to illustrate what I mean (I'm just writing this off the top of my head in github, so please forgive/correct any syntax or other errors)

```suggestion
	// Boolean indicating whether the result is partial (i.e., may
	// change with further processing).
	if result.IsPartial {
	    // Print the partial transcript
	    fmt.Printf("partial: %v...\n", result.Text)
	} else {
	    // The transcript
	    fmt.Printf("final: %v\n", result.Text)

  	    // The confidence the ASR system has in the given transcript,
	    // given as a value between 0 and 1.0, with 1.0 being the most
	    // confident in the result. Note that partial results don't
	    // have an accurate confidence score, so only printing confidence for the final result.
	    fmt.Printf("confidence: %v\n\n", result.Confidence)
        }

```

File: docs-src/content/protobuf/_index.md
Comment (renaissanceGeek) [inline docs-src/content/protobuf/_index.md:34]: Does the first message include the *Cubic* model id or the *Diatheke* model id?  I thought the Cubic model for transcription was set in the Diatheke config file.

File: docs-src/content/protobuf/_index.md
Comment (renaissanceGeek) [inline docs-src/content/protobuf/_index.md:85]:
```suggestion
| transcribe | [TranscribeAction](#TranscribeAction) |  | <p>The client app should call the Transcribe method to capture user input.</p> |
```

File: docs-src/content/protobuf/_index.md
Comment (renaissanceGeek) [inline docs-src/content/protobuf/_index.md:359]:
```suggestion
This action indicates that the client application should call the Transcribe method to
capture the user's input.
```

File: docs-src/content/protobuf/_index.md
Comment (renaissanceGeek) [inline docs-src/content/protobuf/_index.md:393]:
```suggestion
The result from the Transcribe stream. Usually, several partial
```

File: docs-src/content/protobuf/_index.md
Comment (renaissanceGeek) [inline docs-src/content/protobuf/_index.md:401]:
```suggestion
| confidence | [double](#double) |  | <p>Confidence estimate between 0 and 1. A higher number represents a higher likelihood that the transcription is correct.</p> |
```

File: docs-src/content/protobuf/_index.md
Comment (renaissanceGeek) [inline docs-src/content/protobuf/_index.md:402]:
```suggestion
| is_partial | [bool](#bool) |  | <p>True if this is a partial result, in which case the next result will be for the same audio, either repeating or correcting the text in this result. When false, this represents the final transcription for an utterance, which will not change with further audio input. It is sent when the ASR has identified an endpoint. After the final transcription is sent, any additional results sent on the Transcribe stream belong to the next utterance.</p> |
```

**Review by renaissanceGeek — APPROVED** [review]
I left a few suggestions, but overall looks good!

File: docs-src/content/protobuf/_index.md
Comment (mplooster) [reply docs-src/content/protobuf/_index.md:402]: Also auto-generated. I suppose I can go back an update the protobuf file and regenerate.

File: docs-src/content/protobuf/_index.md
Comment (mplooster) [reply docs-src/content/protobuf/_index.md:34]: FYI, this was auto-generated from the protobuf file.

In the server's current implementation, it is expecting the Cubic model ID to be there, so it is correct as written. The fact that there is an associate created between a Diatheke model and a Cubic model in the server config file is significant for generating a TranscribeAction (which the server will populate with both the Cubic and Diatheke model ID), but not required for this stream. This allows a client application to create a Transcribe stream with just a Cubic model ID at any time they want (maybe in response to a button click, for example), without it being prompted by a TranscribeAction in the session output. They could even add the Diatheke model ID too, to get the stream-end conditions if they want it, and it doesn't have to be the same association as defined in the server config file. Think of it more like this:

* Cubic Model ID = Always used for ASR, regardless of optional Diatheke model.
* Diatheke Model ID = Optional, but always used to to define the end-stream conditions.

**Comment by kapitoshka438** [comment]
I've added [PR](https://github.com/cobaltspeech/sdk-diatheke/pull/65) to your branch with updated Swift part.

**Comment by mplooster** [comment]
> I've added PR to your branch with updated Swift part.

Thanks! I won't merge this until your PR is merged with mine.

**Comment by kapitoshka438** [comment]
> > I've added PR to your branch with updated Swift part.
> 
> Thanks! I won't merge this until your PR is merged with mine.

I've made changes based on your comments and also regenerated swift pb and grpc files.

**Review by kapitoshka438 — APPROVED** [review]
Great

**Pull request merged**
